runsc/container/container.go

```diff
@@ -697,12 +698,13 @@ func (c *Container) ForwardSignals(pid int32, fgProcess bool) func() {
 
 // Checkpoint sends the checkpoint call to the container.
 // The statefile will be written to f, the file at the specified image-path.
-func (c *Container) Checkpoint(f *os.File) error {
+// The statefile compression is determined by the state of the flags argument.
```

Comment:
After the above comment, this nomenclature would no longer be adequate: this argument is just passing in compression information, not a whole set of flags.

So two options here:
- Convert `statefile.Flags` to a `struct` in `statefile` where the compression level is one (and the only) field. (New options that affect checkpoint format would go into that `struct`.)

```go
// statefile.go:
type CompressionLevel string
const (
  CompressionLevelFlateBestSpeed = "flate-best-speed"
  CompressionLevelNone = "none"
)
type Options struct {
  Compression CompressionLevel
}

// container.go:
func (c *Container) Checkpoint(cid string, f *os.File, options statefile.Options) error { ... }

// sandbox.go:
func (s *Sandbox) Checkpoint(f *os.File, options statefile.Options) error { ... }
```

- Rename this argument to `compressionLevel` or similar to better reflect what it means. (New options that affect checkpoint format would be added as additional arguments.)

```go
// statefile.go:
type CompressionLevel string
const (
  CompressionLevelFlateBestSpeed = "flate-best-speed"
  CompressionLevelNone = "none"
)

// container.go:
func (c *Container) Checkpoint(cid string, f *os.File, compression statefile.CompressionLevel) error { ... }

// sandbox.go:
func (s *Sandbox) Checkpoint(f *os.File, compression statefile.CompressionLevel) error { ... }
```

Same thing for `sandbox.Checkpoint` in `sandbox.go`.

Reply:
yeah, agree. I decided to change it to `statefile.Options` as I expect there will be changes in the future and I don't want others to extend Checkpoint() functions with other optional arguments.